### PR TITLE
Snfe psrcat plot

### PR DESCRIPTION
**Pull request by keegansmith21**
Added a mode to SNFE that plots pulsars SED as per ATNF data

### CODE REVIEW

File: utils/sn_flux_est.py [hunk 1]
```diff
@@ -824,18 +955,28 @@ def est_pulsar_sn(pulsar, obsid,\
...
+            raj = None
+            decj = None
+        else:
+            raj = args.pointing.split("_")[0]
+            decj = args.pointing.split("_")[1]
+
+        SN, SN_err = est_pulsar_sn(pulsar, args.obsid,\
+                beg=args.beg, end=args.end, p_ra=raj, p_dec=decj, plot_flux=args.plot_est, query=query)
+    elif args.mode == "ATNF":
+        if args.pulsar==None:
```
Comment (NickSwainston) [inline utils/sn_flux_est.py]: Codacy Issue found: [comparison to None should be 'if cond is None:'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)

File: utils/sn_flux_est.py [hunk 2]
```diff
@@ -824,18 +967,27 @@ def est_pulsar_sn(pulsar, obsid,\
...
+            raj = None
+            decj = None
+        else:
+            raj = args.pointing.split("_")[0]
+            decj = args.pointing.split("_")[1]
+
+        SN, SN_err = est_pulsar_sn(pulsar, args.obsid,\
+                beg=args.beg, end=args.end, p_ra=raj, p_dec=decj, plot_flux=args.plot_est, query=query, min_z_power=args.min_z_power)
+    elif args.mode == "ATNF":
+        if args.pulsar==None:
```
Comment (NickSwainston) [inline utils/sn_flux_est.py]: Codacy Issue found: [comparison to None should be 'if cond is None:'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)

File: utils/sn_flux_est.py [hunk 3]
```diff
@@ -178,6 +228,30 @@ def pulsar_beam_coverage(obsid, pulsar, beg=None, end=None, ondisk=False, min_po
 
     return enter_files, exit_files
 
+#---------------------------------------------------------------
+def ATNF_data_plot(pulsar_list):
+    """
+    Given a list of pulsars, plots the data for each one from ATNF
```
Comment (NickSwainston) [inline utils/sn_flux_est.py]: This is vague

File: utils/sn_flux_est.py [hunk 4]
```diff
@@ -804,16 +938,23 @@ def est_pulsar_sn(pulsar, obsid,\
...
     parser.add_argument("-e", "--end", type=int, default=None, help="The end time of observation.\
                         If None, will use the end given by a metadata call. Default: None")
     parser.add_argument("--pointing", type=str, default=None, help="The pointing of the target in the format '12:34:56_98:76:54'.\
                         If None, will obtain from a call to ATNF. Default: None")
+    parser.add_argument("min_z_power", type=float, default=0.3, help="The minimum zenith normalised power used to determine if the pulsar\
+                        is in the beam or not")
     parser.add_argument("--plot_est", action="store_true", help="Use this tag to create a plot of flux estimation.")
+    parser.add_argument("--mode", type=str, help="""MODES: 'SNFE' = Estimate S/N and flux for a single pulsar in an obsid\n
+
+                                                'ATNF' = Plot the SED for any number of pulsars using data from ATNF.""")
```
Comment (NickSwainston) [inline utils/sn_flux_est.py]: Make this clearer by expanding SED and explaining that it is not doing any flux calculations

File: utils/sn_flux_est.py (hunk 3 above)
Comment (keegansmith21) [reply utils/sn_flux_est.py]: your comment is also vague

File: utils/sn_flux_est.py (hunk 4 above)
Comment (keegansmith21) [reply utils/sn_flux_est.py]: Why would the user think that it's doing any flux calculations in this mode?

File: utils/sn_flux_est.py (hunk 3 above)
Comment (NickSwainston) [reply utils/sn_flux_est.py]: What data is it plotting?

Comment (keegansmith21) [reply utils/sn_flux_est.py]: Updated

File: database/submit_to_database.py [hunk 5]
```diff
@@ -193,15 +193,22 @@ def flux_cal_and_submit(time_detection, time_obs, metadata, bestprof_data,
...
-    u_w_equiv_bins =  prof_dict["Weq_e"]
-    w_equiv_ms = period/num_bins * w_equiv_bins
-    u_w_equiv_ms = period/num_bins * u_w_equiv_bins
-    scattering = prof_dict["Wscat"]*period/num_bins
-    u_scattering = prof_dict["Wscat_e"]*period/num_bins
-    scattered = prof_dict["scattered"]
+
+    if not prof_dict:
+        logger.info("Profile couldn't be fit. Using old style of profile analysis")
+        sn, u_sn, flags, w_equiv_bins, u_w_equiv_bins, w_equiv_ms, u_w_equiv_ms, scattering, u_scattering, scattered = prof_utils.analyse_pulse_prof(profile, period, verbose=True)
```
Comment (NickSwainston) [inline database/submit_to_database.py]: Codacy Issue found: [line too long (179 > 159 characters)](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)

File: utils/prof_utils.py [hunk 6]
```diff
@@ -616,6 +616,126 @@ def est_sn_from_prof(prof_data, period, alpha=3.):
...
+    shift = -int(np.argmax(prof_data))+int(nbins)//2
+    prof_data = np.roll(prof_data, shift)
+
+    #find sigma and check if profile is scattered
+    sigma, flags = submit_to_database.sigmaClip(prof_data, alpha=3., tol=0.01, ntrials=100)
+    bot_prof_min = (max(prof_data) - min(prof_data)) * .1 + min(prof_data)
+    scattered=False
+    if (np.nanmin(flags) > bot_prof_min) or ( not np.isnan(flags).any() ):
+        logger.warning("The profile is highly scattered. S/N estimate cannot be calculated")
+        if verbose == True:
```
Comment (NickSwainston) [inline utils/prof_utils.py]: Codacy Issue found: [comparison to True should be 'if cond is True:' or 'if cond:'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)

File: utils/prof_utils.py [hunk 7]
```diff
@@ -616,6 +616,126 @@ def est_sn_from_prof(prof_data, period, alpha=3.):
...
+        VERBOSE - when true, the profile is highly scattered
+    """
+
+    nbins = len(prof_data)
+    #centre the profile around the max
+    shift = -int(np.argmax(prof_data))+int(nbins)//2
+    prof_data = np.roll(prof_data, shift)
+
+    #find sigma and check if profile is scattered
+    sigma, flags = submit_to_database.sigmaClip(prof_data, alpha=3., tol=0.01, ntrials=100)
```
Comment (NickSwainston) [inline utils/prof_utils.py]: Codacy Issue found: [undefined name 'submit_to_database'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)

File: database/submit_to_database.py (hunk 5 above)
Comment (NickSwainston) [inline database/submit_to_database.py]: Codacy Issue found: [Unused variable 'flags'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)

File: utils/prof_utils.py [hunk 8]
```diff
@@ -616,6 +616,126 @@ def est_sn_from_prof(prof_data, period, alpha=3.):
...
+        u_simga = sigma / np.sqrt(2 * non_pulse_bins - 2)
+
+        #now calc S/N
+        sn = max(prof_data)/sigma
+        u_sn = sn * np.sqrt(u_prof/max(prof_data)**2 + (u_simga/sigma)**2)
+
+    if verbose==False:
+        return [sn, u_sn]
+
+    elif scattered==False:
```
Comment (NickSwainston) [inline utils/prof_utils.py]: Codacy Issue found: [comparison to False should be 'if cond is False:' or 'if not cond:'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)

File: utils/prof_utils.py [hunk 9]
```diff
@@ -616,6 +616,126 @@ def est_sn_from_prof(prof_data, period, alpha=3.):
...
+                u_p = np.sqrt(u_p**2 + u_prof**2)
+            else:
+                non_pulse_bins += 1
+        u_simga = sigma / np.sqrt(2 * non_pulse_bins - 2)
+
+        #now calc S/N
+        sn = max(prof_data)/sigma
+        u_sn = sn * np.sqrt(u_prof/max(prof_data)**2 + (u_simga/sigma)**2)
+
+    if verbose==False:
```
Comment (NickSwainston) [inline utils/prof_utils.py]: Codacy Issue found: [comparison to False should be 'if cond is False:' or 'if not cond:'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)

**Pull request merged**